### PR TITLE
Automatically create Zapprfile

### LINKED ISSUE #326: Automatically create .zapper.yaml
It would be great to automatically create a Pull Request to add a basic `.zappr.yaml` file to the repository. 

[Creating](https://developer.github.com/v3/repos/contents/#create-a-file) a file and [opening a Pull Request](https://developer.github.com/v3/pulls/#create-a-pull-request) are both easy to do through the API. The added benefit of doing this is that **Zappr will already be in the list of recent status checks** when configuring Protected Branches.

### PR DESCRIPTION
**Pull request by prayerslayer**
Fixes #326 

## Done when

- [x] Pull Request with Zapprfile is only created once

- [x] ...and only if there is no Zapprfile already. If it is, we assume familiarity with Zappr.

- [x] Documentation added

- [x] Tests are present

- [x] Database migration is tested also with old version to avoid downtime

### CODE REVIEW

File: server/handler/RepositoryHandler.js [hunk 1]
```diff
@@ -12,6 +12,11 @@ class RepositoryHandler {
     this.github = github
   }
 
+  onWelcome(id) {
+    debug(`welcome repository ${id}!`)
+    return Repository.findById(id).then(repo => repo.update({welcomed: true}))
```
Comment (prayerslayer) [inline server/handler/RepositoryHandler.js]: This is probably two queries instead of one :(

**Comment by prayerslayer** [comment]
Fixed the conflicts @mfellner, ready to review ✊

File: docs/setup.md [hunk 2]
```diff
@@ -40,9 +40,9 @@ It is customized by everything under `approvals`. The following options are supp
 * `ignore`: Whether Zappr should ignore approvals by the `last_committer` on the pull request, the `pr_opener`, `both` or `none`. Defaults to `none`.
 * `pattern`: Since approvals are essentially comments that match a pattern, you can configure the pattern! It's a string that will be passed to Javascript's `RegExp` constructor and defaults to `^(:+1:|👍)$`. (Tip: If you're not sure about your regex, [regex101.com](https://regex101.com/) is great to test it.)
 * `from`: By default any comment that matches the pattern is considered an approval, regardless of the author. You can change this by
- * `organization`: list organizations under `orgs` that the author has to be a public member of
- * `usernames`: list usernames under `users`
- * `collaborators`: set the `collaborators` flag to `true`
+  * `organization`: list organizations under `orgs` that the author has to be a public member of
```
Comment (mfellner) [inline docs/setup.md]: Should be 4 spaces I think.

File: server/service/GithubService.js [hunk 3]
```diff
@@ -299,16 +301,144 @@ export class GithubService {
...
+   * https://developer.github.com/v3/repos/contents/#create-a-file
+   *
+   * @param user
+   * @param repo
+   * @param branch
+   * @param path
+   * @param content
+   * @param accessToken
+   */
+  async createFile(user, repo, branch, path, content, accessToken) {
```
Comment (mfellner) [inline server/service/GithubService.js]: Don't need async here (no await).

File: server/service/GithubService.js [hunk 4]
```diff
@@ -299,16 +301,144 @@ export class GithubService {
...
+   *
+   * @param user
+   * @param repo
+   * @param head
+   * @param base
+   * @param title
+   * @param body
+   * @param accessToken
+   */
+  async createPullRequest(user, repo, head, base, title, body, accessToken) {
```
Comment (mfellner) [inline server/service/GithubService.js]: Don't need async here (no await).

**Comment by prayerslayer** [comment]
👍

**Comment by mfellner** [comment]
👍

**Pull request merged**